library/src/blas_ex/rocblas_gemm_ext2.cpp

```diff
@@ -0,0 +1,345 @@
+                              m,
+                              n,
+                              k,
+                              alpha,
+                              a,
+                              a_type_string,
+                              a_row_stride,
+                              a_col_stride,
+                              b,
+                              b_type_string,
+                              b_row_stride,
+                              b_col_stride,
+                              beta,
+                              c,
+                              c_type_string,
+                              c_row_stride,
+                              c_col_stride,
+                              d,
+                              d_type_string,
+                              d_row_stride,
+                              d_col_stride,
+                              compute_type_string,
+                              algo,
+                              solution_index,
+                              flags);
+            }
+        }
+
+        if(layer_mode & rocblas_layer_mode_log_profile)
+        {
+            log_profile(handle,
+                        "rocblas_gemm_ex",
+                        "a_type",
+                        a_type_string,
+                        "b_type",
+                        b_type_string,
+                        "c_type",
+                        c_type_string,
+                        "d_type",
+                        d_type_string,
+                        "compute_type",
+                        compute_type_string,
+                        "M",
+                        m,
+                        "N",
+                        n,
+                        "K",
+                        k,
+                        "a_row_stride",
+                        a_row_stride,
+                        "a_col_stride",
+                        a_col_stride,
+                        "b_row_stride",
+                        b_row_stride,
+                        "b_col_stride",
+                        b_col_stride,
+                        "c_row_stride",
+                        c_row_stride,
+                        "c_col_stride",
+                        c_col_stride,
+                        "d_row_stride",
+                        d_row_stride,
+                        "d_col_stride",
+                        d_col_stride,
+                        "algo",
+                        algo,
+                        "solution_index",
+                        solution_index,
+                        "flags",
+                        flags);
+        }
+    }
+
+    // quick return m,n,k equal to 0 is valid in BLAS
+    // Note: k==0 is not a quick return, because C still has to be multiplied by beta
+    if(!m || !n)
+        return rocblas_status_success;
+
+    // sizes must not be negative
+    if(m < 0 || n < 0 || k < 0)
+        return rocblas_status_invalid_size;
```

Comment:
Legacy BLAS has size checks like m<0 before quick return like !m . Can you move the size checks before the quick returns.